src/lib/viewers/doc/PresentationViewer.js

```diff
@@ -42,6 +41,9 @@ class PresentationViewer extends DocBaseViewer {
         // Set up preloader
         this.preloader = new PresentationPreloader(this.previewUI, { api: this.api });
         this.preloader.addListener('preload', this.onPreload.bind(this));
+
+        // listen for set page events
+        this.addListener('setpage', this.setPage);
```

Comment:
Can we just call this method on the viewer itself? Or proxy it in Preview.js?

Reply:
In the current implementation, I am emitting this from `box-annotations` with the page from the annotation location.